app/src/main/java/com/example/tawazz/storage/Storage.java

```diff
@@ -0,0 +1,55 @@
+package com.example.tawazz.storage;
+
+import android.content.Context;
+import android.net.Uri;
+import android.widget.ImageView;
+
+import com.example.tawazz.download.UriFilesDownloader;
+import com.example.tawazz.storage.exceptions.FailedStoreException;
+import com.example.tawazz.task.TaskCompletedWaiterFactory;
+import com.example.tawazz.utils.exceptions.FailedWaitingForConditionException;
+import com.example.tawazz.utils.firebase.GlideApp;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+public class Storage {
+    private StorageReference mStorageReference;
+    private TaskCompletedWaiterFactory mTaskCompletedWaiterFactory;
+
+    public Storage(StorageReference storageReference, TaskCompletedWaiterFactory taskCompletedWaiterFactory) {
+        this.mStorageReference = storageReference;
+        this.mTaskCompletedWaiterFactory = taskCompletedWaiterFactory;
+    }
+
+    public UploadTask store(Uri srcDir, Uri destDir) throws FailedStoreException {
+        StorageReference userIconStorageRef = mStorageReference.child(destDir.getPath());
+        return userIconStorageRef.putFile(srcDir);
+    }
+
+    public void safeStore(Uri srcDir, Uri destDir) throws FailedStoreException {
+        try {
+            UploadTask uploadTask = store(srcDir, destDir);
+            mTaskCompletedWaiterFactory.create(uploadTask).waitTill();
+        } catch (FailedWaitingForConditionException e) {
+            throw new FailedStoreException(e);
+        }
+    }
+
+//    (!) unused
```

Comment:
Since you removed the `mDownloadInvoker` from here, this unused function (`get`) is no longer valid, I suggest that you delete it. You can easily recover it from git history if needed in the future.